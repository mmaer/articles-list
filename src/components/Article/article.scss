@import '../../styles/variable.scss';

.article {
  display: flex;
  margin: 10px 5px;

  &__image {
    margin-right: 20px;
    width: 33%;

    img {
      max-width: 100%;
      max-height: 100%;
    }
  }

  &__content {
    display: flex;
    flex-direction: column;
    width: 66%;
  }

  &__title-date {
    flex-direction: column;
    display: flex;
    justify-content: space-between;
    margin-bottom: 20px;

    @media (#{$above-tablet}) {
      flex-direction: row;
    }
  }

  &__title {
    font-size: 26px;
    margin-bottom: 10px;

    @media (#{$above-tablet}) {
      flex-shrink: 3000;
      margin-bottom: 0;
    }
  }

  &__date {
    @media (#{$above-tablet}) {
      align-self: flex-end;
    }
  }

  &__text {
    display: none;

    @media (#{$above-tablet}) {
      display: block;
      width: 100%;
    }
  }
}
