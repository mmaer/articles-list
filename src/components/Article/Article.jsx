import React from 'react';

import imageNotFound from '../../images/image-not-found.png'

import './article.scss';

const Article = ({ date, image, title, premable }) => {
  return (
    <div className="article">
      <div className="article__image">
        <img src={image ? image : imageNotFound} alt="nature" />
      </div>
      <div className="article__content">
        <div className="article__title">
          <h2>{title}</h2>
        </div>
        <div className="article__date">{date}</div>
        <div className="article__text">{premable}</div>
      </div>
    </div>
  );
};

export default Article;
