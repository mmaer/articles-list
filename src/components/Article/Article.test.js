import React from 'react';
import { shallow } from 'enzyme';

import Article from './';

const fakeArticle = {
  date: '2. februar 2019',
  image: 'https://placeimg.com/280/180/nature',
  title: 'Vålerengas førsterekke smadrer rivalene: - Seriegullet er I våre hender',
  preamble:
    'MERÅKER (VG) Finn-Hågen Krogh (28) opplevde den gedigne nedturen da han ble vraket til OL-sprinten i Sotsji etter at han først var tatt ut på laget. Nå føler han seg aldri trygg på å få starte i mesterskap.',
};

describe('Article component', () => {
  it('Should match snapshot', () => {
    const wrapper = shallow(<Article {...fakeArticle} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should match snapshot when image is empty', () => {
    const wrapper = shallow(<Article {...fakeArticle} image="" />);
    expect(wrapper).toMatchSnapshot();
  });
});
