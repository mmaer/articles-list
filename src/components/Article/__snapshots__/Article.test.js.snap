// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article component Should match snapshot 1`] = `
<div
  className="article"
>
  <div
    className="article__image"
  >
    <img
      alt="Vålerengas førsterekke smadrer rivalene: - Seriegullet er I våre hender"
      src="https://placeimg.com/280/180/nature"
    />
  </div>
  <div
    className="article__content"
  >
    <div
      className="article__title-date"
    >
      <div
        className="article__title"
      >
        <h2>
          Vålerengas førsterekke smadrer rivalene: - Seriegullet er I våre hender
        </h2>
      </div>
      <div
        className="article__date"
      >
        2. februar 2019
      </div>
    </div>
    <div
      className="article__text"
    >
      MERÅKER (VG) Finn-Hågen Krogh (28) opplevde den gedigne nedturen da han ble vraket til OL-sprinten i Sotsji etter at han først var tatt ut på laget. Nå føler han seg aldri trygg på å få starte i mesterskap.
    </div>
  </div>
</div>
`;

exports[`Article component Should match snapshot when image is empty 1`] = `
<div
  className="article"
>
  <div
    className="article__image"
  >
    <img
      alt="Vålerengas førsterekke smadrer rivalene: - Seriegullet er I våre hender"
      src=""
    />
  </div>
  <div
    className="article__content"
  >
    <div
      className="article__title-date"
    >
      <div
        className="article__title"
      >
        <h2>
          Vålerengas førsterekke smadrer rivalene: - Seriegullet er I våre hender
        </h2>
      </div>
      <div
        className="article__date"
      >
        2. februar 2019
      </div>
    </div>
    <div
      className="article__text"
    >
      MERÅKER (VG) Finn-Hågen Krogh (28) opplevde den gedigne nedturen da han ble vraket til OL-sprinten i Sotsji etter at han først var tatt ut på laget. Nå føler han seg aldri trygg på å få starte i mesterskap.
    </div>
  </div>
</div>
`;
